pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -299,6 +300,11 @@ def plot_group(keys, values, ax):
         if fontsize is not None:
             ax.tick_params(axis="both", labelsize=fontsize)
         if kwds.get("vert", 1):
+            ticks = ax.get_xticks()
```

Comment:
This comes from a change in MPL being stricter about `set_xticks`. This worked for matplotlib 3.2

```python
import pandas as pd
import matplotlib.pyplot as plt
import string
import numpy as np

fig, (ax1, ax2) = plt.subplots(ncols=2, sharex=True)
ax1.boxplot([
    np.array([1, 2, 3, 4]),
    np.array([1, 2, 3, 4])
])
ax2.boxplot([
    np.array([1, 2, 2, 3]),
    np.array([1, 2, 3, 4])
])

ax1.set_xticklabels(['A', 'B'])
```

We set 2 x ticklabels despite there being 4.

Reply:
Yes, this was a deliberate change... https://github.com/matplotlib/matplotlib/pull/17266  The obvious problem if you only give us 2 tick labels is that its ambiguous which of the 4 ticks you wanted labeled.  

MPL now has categorical axes which I think were added because pandas has them.  Probably too big a project to homogenize, but....